src/modules/personenkontext/specification/befristung-required-bei-rolle-befristungspflicht.ts

```diff
@@ -0,0 +1,26 @@
+import { RolleRepo } from '../../rolle/repo/rolle.repo.js';
+import { RollenMerkmal } from '../../rolle/domain/rolle.enums.js';
+import { Personenkontext } from '../domain/personenkontext.js';
+import { Rolle } from '../../rolle/domain/rolle.js';
+
+export class CheckBefristungSpecification {
+    public constructor(private readonly rolleRepo: RolleRepo) {}
+
+    public async checkBefristung(sentPKs: Personenkontext<boolean>[]): Promise<boolean> {
+        // Extract unique Rolle IDs from sentPKs
+        const uniqueRolleIds: Set<string> = new Set(sentPKs.map((pk: Personenkontext<boolean>) => pk.rolleId));
+
+        const mapRollen: Map<string, Rolle<true>> = await this.rolleRepo.findByIds(Array.from(uniqueRolleIds));
+
+        for (const pk of sentPKs) {
+            const rolle: Rolle<true> | undefined = mapRollen.get(pk.rolleId);
+            if (rolle && rolle.merkmale.includes(RollenMerkmal.BEFRISTUNG_PFLICHT)) {
+                // Check if befristung is set
+                if (pk.befristung === undefined) {
+                    return false;
+                }
+            }
+        }
```

Comment:
This check would allow a `befristung` to always be present, even if none of the roles require it. Is this correct?
In that case, you could do an early return before fetching the `rollen` from the database, to save the trip to the DB (`if (pk.befristung) return true;`).